ste/uploader-blobFS.go

```diff
@@ -169,22 +175,44 @@ func (u *blobFSUploader) GenerateUploadFunc(id common.ChunkID, blockIndex int32,
 	})
 }
 
+var flushThreshold int64 = (1024 * 1024 * 1024) * 20 //20GB
+//TODO: Should a flag be present for the threshold?
+var thresholdSetup bool
+
+func setupThreshold(blockSize int64) {
+	//Find the closest multiple of the block size to the flush threshold
+	m := flushThreshold % blockSize
+
+	if math.Abs(float64(m)) < math.Abs(float64((flushThreshold-m+blockSize)-flushThreshold)) {
+		flushThreshold -= m
+	} else {
+		flushThreshold = (flushThreshold - m) + blockSize
+	}
+}
+
 func (u *blobFSUploader) Epilogue() {
 	jptm := u.jptm
 
 	// flush
 	if jptm.TransferStatus() > 0 {
 		md5Hash, ok := <-u.md5Channel
+		ss := jptm.Info().SourceSize
```

Comment:
Minor (and probably don't bother fixing it in this case). Stylistically, we'd normally reverse the order of the md5Hash and ss := lines, so that the if ok test was immediately after the thing that sets ok.